data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+         -nocase "mistake|wrong|fault|error|fail|exception|not*found|timed*out" {puts "Can not login virsh console to \${guest_domain}\r"; set ret_result 1}
+      }
+
+      if { \${ret_result} == 0 } {
+         set timeout 600
+         expect "~ #"
+         send "mkdir -p \${logs_folder};export time_stamp=\`date \'+%Y%m%d%H%M%S\'\`\r"
+         expect "~ #"
+         if { \${extra_logs} == {support_config} } {
+            send "rm -f -r \${logs_folder}/*supportconfig*\r"
+            send "supportconfig -t \${logs_folder} -B guest_\${guest_transformed}_supportconfig_\\\${time_stamp}\r"
+         }
+         if { \${extra_logs} != {support_config} && \${extra_logs} != "" } {
+            send "mkdir -p \${logs_folder}/guest_\${guest_transformed}_extra_logs;cp -r -f \${extra_logs} \${logs_folder}/guest_\${guest_transformed}_extra_logs\r"
+         }
+         expect {
+            "~ #" {send "export ret=\\\$?;cd /tmp;(if \[\[ \\\$ret -ne 0 ]];then printenv USER;fi)\r"; exp_continue -continue_timer}
+            "root" {puts "Can not collect \${extra_logs} from \${guest_domain}\r"; set ret_result 1;  exp_continue -continue_timer}
+            "tmp #" {send "cd ~\r"; send "exit\r"}
+         }
+      }
+
+      if { \${ret_result} == 0 && \${extra_logs} != "" } {
+         expect -nocase "login: $" {puts "Successfully collected \${extra_logs} from \${guest_domain}\r"; exit \${ret_result}}
+      }
+
+      if { \${ret_result} == 1 && \${extra_logs} != "" } {
+         expect -nocase "login: $" {puts "Will try to collect \${extra_logs} from \${guest_domain} again\r"}
+      }
+
+      if { \${extra_logs} == "" } {
+         expect -nocase "login: $" {puts "Nothing to collect from \${guest_domain}. Please specify something.\r"}
+      }
+
+      set retry_times [expr \${retry_times}-1]
+      close -i \${spawn_id}
+}
+exit \${ret_result}
+EOF
+
+        echo -e "${expfile} to be executed is as below:"
+        cat ${expfile}
+  
+	for procid in `ps aux | grep  ${expfile} | grep -v "grep" | awk '{print $2}'`;do
+            kill -9 ${procid}
+	done
+	for procid in `ps aux | grep "virsh console" | grep -v "grep" | awk '{print $2}'`;do
+	    kill -9 ${procid}
+	done
+	echo -e "expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}""
+	expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}"
+        if [[ $? == 0 ]];then
+           echo -e "${expfile} returned with success. Successfully collected ${extra_logs} via ${guest_domain} console."
+           rm -f ${expfile}
+	   return 0
+        else
+           echo -e "${expfile} returned with failure. Failed to collect ${extra_logs} via ${guest_domain} console."
+           rm -f ${expfile}
+           return 1
+	fi
+}
+
+#Collect supportconfig from both host and guest. Will resort to guest virsh console if collecting from ssh failed.
+function collect_supportconfig() {
```

Comment:
Seems this function is designed to collect log from either host or guest, not both at the same time, right? Besides it seems it supports "some" combination of parameter 1,2,3,4. 
Can you give the clarification and the typical supported usage example  in the function description comment? It helps easier usage for others.

Reply:
It supports collecting supportconfig from both host and  guest. I do not know what do you mean by "at the same time". There is no "at the same time" because there is only one process running. Collecting must happen one by one but you only need one function collect_supportconfig. 
Usage:
For host: collect_supportconfig logfolder 'host'     (only support collecting from localhost)
For guest: collect_supportconfig logfolder 'guest' guestip guestdomain

All these parameter are necessary.

Reply:
By at "the same time", I mean call this function one time , both host and guest  logs are collected.  This is a question I ask actually because function description in comment "#Collect supportconfig from both host and guest." gives the fake feeling that it can do this. But actually it can not. So this is the point I mean. Suggest to not blur this in function description comment and add typical usage example there.

Reply:
You can have different understanding. But anyway I will help direct you to the very exact meaning of this function.